server/direct/puppetserver-foreman

```diff
@@ -0,0 +1,300 @@
+#!/usr/bin/env python3
+
+from socketserver import ThreadingUnixStreamServer, StreamRequestHandler
+import json
+import logging
+import socket
+
+import requests
+import yaml
+
+SESSION = requests.Session()
+
+# TODO: configuration
+FOREMAN_URL = 'https://HOSTNAME'
+SESSION.auth = ('USERNAME', 'PASSWORD')
+
+
+def load_json_or_yaml(data):
+    try:
+        return json.loads(data)
+    except:
+        try:
+            return yaml.safe_load(data)
+        except:
+            raise ValueError('Invalid JSON/YAML')
```

Comment:
Given any JSON document is valid YAML, and you depend on PyYAML anyways, just always parse with YAML?

Reply:
YAML parsing does have odd quirks so in general I'd prefer to avoid. However, I don't have a good content detection. Perhaps I can check if it starts with `---` and directly move to YAML.